src/java/org/apache/cassandra/journal/Params.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.journal;
+
+public interface Params
```

Comment:
is this meant to be used in `cassandra.yaml`?   If so why not just make this a class with the defaults from `DEFAULT`?

https://issues.apache.org/jira/browse/CASSANDRA-18221 is in the works now, that patch will add configs to Accord and plumb that into C*.  One feature that will exist there is a new annotation of `@Embedded` which will let you have complex/nested types but they get "embedded" into the config yaml...  so in your case I think you want

```
// in Config.java
@Embedded
public final Params accord_journal = new Params;
```

and the yaml would be

```
accord_journal_segment_size: 33554432
accord_journal_failure_policy: STOP
...
```

Reply:
> is this meant to be used in cassandra.yaml? If so why not just make this a class with the defaults from DEFAULT

Not quite. It's supposed to be a level higher than that, not bound to a particular implementation.

Reply:
Also, for now, not yet going to expose any more params other than directory, already exposed. Later, will most likely have to also expose segment size at least, however.